### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by siftal**
- Fixes a bug in the consensus receiver service to wait for confirmations before getting blocks to prevent missing operations in reorgs.
- Updates `unusedSponsorships` function to only count sponsorships that are finalized.
- Updates `GET /verifications/{app}` to be able to return linked ids verified for the app in different periods (last/previous/all).
- Adds transaction hash to the operations documents to simplify tracking operations.

### CODE REVIEW

File: consensus/receiver.py [hunk 1]
```diff
@@ -16,6 +16,8 @@
 if config.INFURA_URL.count('rinkeby') > 0 or config.INFURA_URL.count('idchain') > 0:
     w3.middleware_onion.inject(geth_poa_middleware, layer=0)
 
+NUM_SEALERS = 10
```
Comment (abramsymons) [inline consensus/receiver.py:19]: initialization can be ignored as we are retrying on errors while updating this number.

File: web_services/foxx/brightid/db.js [hunk 2]
```diff
@@ -922,13 +923,18 @@ function isEthereumAddress(address) {
...
     const vel = app.verificationExpirationLength;
     query["roundedTimestamp"] = vel ? parseInt(Date.now() / vel) * vel : 0;
+  } else if (period == "previous") {
+    const vel = app.verificationExpirationLength;
+    query["roundedTimestamp"] = vel
+      ? (parseInt(Date.now() / vel) - 1) * vel
+      : 0;
   }
   for (const verification of app.verifications) {
     query["verification"] = verification;
```
Comment (abramsymons) [inline web_services/foxx/brightid/db.js:940]: I could not place comment in this bellow section:
```
const appUserIds = appIdsColl
      .byExample(query)
      .toArray()
      .map((d) => d.appId);```
I think we should create a `Set` here.

Comment (siftal) [reply web_services/foxx/brightid/db.js:940]: https://github.com/BrightID/BrightID-Node/commit/4b2d386eea24a0cbbdfaa7054d5b90e94a5cdf08

File: web_services/profile/app.js [hunk 3]
```diff
@@ -67,21 +67,16 @@ app.post("/upload/:channelId", function (req, res) {
     channelCache.set(channelId, channel, ttl)
     console.log(`Created new channel ${channelId} with TTL ${channel.ttl}`)
   } else {
-    // existing channel. check if this channel was about to expire, but got another upload
-    if (channel.entries.size === 0) {
-      console.log(`Restoring requested TTL ${channel.ttl} for channel ${channelId}`)
-      channelCache.ttl(channelId, channel.ttl)
-    }
+    // extend TTL of the channel when got an upload
```
Comment (TripleSpeeder) [inline web_services/profile/app.js:70]: This is a general change of channel service logic from having a fixed Time-to-live to a on-demand time to live.
If we do this we need to refactor other parts of the profile service code that become obsolete (e.g. handling of channel becoming empty, finalTTL related changes...)
Also the connection making code in the client is tailored around having a fixed TTL. We need to review implications on the client before releasing this.

Comment (abramsymons) [reply web_services/profile/app.js:70]: we will remove this change from this release to think about it more.
we will follow the approach of using longer ttl for explorer for now.
thank you.

**Pull request merged**